### PR TITLE
[580] Create some integration tests. Create profiles for WildFly to r…

### LINKED ISSUE #580: We need some unit test implementation CI testing
##### Issue Overview
It seems like we need more unit test framework validation in the CI tests

##### Expected Behaviour
We should validate key versions of Junit 4/5 and TestNG

##### Current Behaviour
We don't have such tests

### PR DESCRIPTION
**Pull request by jamezp**
…un the tests.

#### Short description of what this resolves:

Introduce some integration tests.

#### Changes proposed in this pull request:

- Create integration tests to run against different implementations.
- The first implementation to test with and enabled in this PR is WildFly

Fixes #580

### CODE REVIEW

**Comment by lprimak** [comment]
See https://github.com/flowlogix/base-pom the way I did this for Payara so that profiles are not repeated for every project.
Also see https://github.com/flowlogix/depchain/tree/main/integration-test for how I automatically integrate testcontainers, and remote / local / k8s payara in a common, easy-to-use way

**Comment by starksm64** [comment]
@jamezp What about not having this inherrit the arquillian-parent pom and only use the bom? The reason is that it can be used to test any version of arquillian-core. For example, I have a local 1.9.0.Final-SNAPSHOT build and I want to run these test against it, but I have to go through all of the poms and update the version. If it was just importing the bom and using a property, it could be retargeted using a command line override.

**Comment by starksm64** [comment]
I guess I can already override the version on the command line.

**Comment by jamezp** [comment]
Yeah, you should be able to override it. It definitely doesn't need to have the parent POM for sure. I'll make sure to set it up to allow all dependencies to be overridden so we can test in CI too.

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+name: Arquillian Integration Tests
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
```
Comment (jamezp) [inline .github/workflows/integration-tests.yml:12]: This will add 6 jobs to each PR. I'm not sure if we want to do that, but welcome feedback.

**Comment by jamezp** [comment]
These Windows failures are strange. The BOM is supposed to build as the last module before the integration tests. However, it looks like it's being build second:
```
2024-07-08T20:21:01.5961326Z [INFO] ----------------< org.jboss.arquillian:arquillian-bom >-----------------
2024-07-08T20:21:01.5962746Z [INFO] Building Arquillian BOM 1.9.0.Final-SNAPSHOT                      [2/45]
2024-07-08T20:21:01.5963895Z [INFO] --------------------------------[ pom ]---------------------------------
```

The reactor itself looks correct though:
```
2024-07-08T20:22:06.3671544Z [INFO] Reactor Summary for Arquillian Aggregator 1.9.0.Final-SNAPSHOT:
2024-07-08T20:22:06.3672302Z [INFO] 
2024-07-08T20:22:06.3672911Z [INFO] Arquillian Aggregator .............................. SUCCESS [ 15.733 s]
2024-07-08T20:22:06.3674465Z [INFO] Arquillian Build ................................... SUCCESS [  1.109 s]
2024-07-08T20:22:06.3676125Z [INFO] Arquillian Core API ................................ SUCCESS [  9.131 s]
2024-07-08T20:22:06.3677201Z [INFO] Arquillian Core SPI ................................ SUCCESS [  1.047 s]
2024-07-08T20:22:06.3678309Z [INFO] Arquillian Core Implementation Base ................ SUCCESS [  3.611 s]
2024-07-08T20:22:06.3679413Z [INFO] Arquillian Core Aggregator ......................... SUCCESS [  0.094 s]
2024-07-08T20:22:06.3680495Z [INFO] Arquillian Config API .............................. SUCCESS [  0.438 s]
2024-07-08T20:22:06.3681881Z [INFO] Arquillian Config SPI .............................. SUCCESS [  0.265 s]
2024-07-08T20:22:06.3682903Z [INFO] Arquillian Test API ................................ SUCCESS [  0.297 s]
2024-07-08T20:22:06.3683936Z [INFO] Arquillian Test SPI ................................ SUCCESS [  1.907 s]
2024-07-08T20:22:06.3684972Z [INFO] Arquillian Test Implementation Base ................ SUCCESS [  2.314 s]
2024-07-08T20:22:06.3686018Z [INFO] Arquillian Config Implementation Base .............. SUCCESS [  6.051 s]
2024-07-08T20:22:06.3687289Z [INFO] Arquillian Config Aggregator ....................... SUCCESS [  0.031 s]
2024-07-08T20:22:06.3688492Z [INFO] Arquillian Test Aggregator ......................... SUCCESS [  0.047 s]
2024-07-08T20:22:06.3689865Z [INFO] Arquillian Container SPI ........................... SUCCESS [  1.344 s]
2024-07-08T20:22:06.3690995Z [INFO] Arquillian Container Implementation Base ........... SUCCESS [  2.782 s]
2024-07-08T20:22:06.3692524Z [INFO] Arquillian Container Test API ...................... SUCCESS [  0.547 s]
2024-07-08T20:22:06.3693929Z [INFO] Arquillian Container Test SPI ...................... SUCCESS [  0.454 s]
2024-07-08T20:22:06.3695336Z [INFO] Arquillian Container Test Implementation Base ...... SUCCESS [  3.688 s]
2024-07-08T20:22:06.3696834Z [INFO] Arquillian Container Aggregator .................... SUCCESS [  0.062 s]
2024-07-08T20:22:06.3697934Z [INFO] Arquillian TestRunner JUnit Core ................... SUCCESS [  3.245 s]
2024-07-08T20:22:06.3699094Z [INFO] Arquillian TestRunner JUnit Standalone ............. SUCCESS [  1.720 s]
2024-07-08T20:22:06.3699737Z [INFO] Arquillian TestRunner JUnit Container .............. SUCCESS [  1.750 s]
2024-07-08T20:22:06.3701058Z [INFO] Arquillian TestRunner JUnit Aggregator ............. SUCCESS [  0.016 s]
2024-07-08T20:22:06.3702365Z [INFO] Arquillian TestRunner JUnit 5 Core ................. SUCCESS [  0.594 s]
2024-07-08T20:22:06.3703520Z [INFO] Arquillian TestRunner JUnit 5 Container ............ SUCCESS [  2.751 s]
2024-07-08T20:22:06.3704645Z [INFO] Arquillian TestRunner JUnit Jupiter Aggregator ..... SUCCESS [  0.063 s]
2024-07-08T20:22:06.3705778Z [INFO] Arquillian TestRunner TestNG Core .................. SUCCESS [  2.424 s]
2024-07-08T20:22:06.3707320Z [INFO] Arquillian TestRunner TestNG Standalone ............ SUCCESS [  0.406 s]
2024-07-08T20:22:06.3708434Z [INFO] Arquillian TestRunner TestNG Container ............. SUCCESS [  1.610 s]
2024-07-08T20:22:06.3709568Z [INFO] Arquillian TestRunner TestNG Aggregator ............ SUCCESS [  0.031 s]
2024-07-08T20:22:06.3710847Z [INFO] Arquillian TestEnricher CDI ........................ SUCCESS [  3.206 s]
2024-07-08T20:22:06.3712512Z [INFO] Arquillian TestEnricher EJB ........................ SUCCESS [  1.203 s]
2024-07-08T20:22:06.3713621Z [INFO] Arquillian TestEnricher Resource ................... SUCCESS [  1.031 s]
2024-07-08T20:22:06.3714945Z [INFO] Arquillian TestEnricher InitialContext ............. SUCCESS [  0.235 s]
2024-07-08T20:22:06.3716022Z [INFO] Arquillian TestEnricher Aggregator ................. SUCCESS [  0.015 s]
2024-07-08T20:22:06.3717216Z [INFO] Arquillian Protocol Servlet 2.5/3.x ................ SUCCESS [  3.845 s]
2024-07-08T20:22:06.3717893Z [INFO] Arquillian Protocol JMX ............................ SUCCESS [  1.517 s]
2024-07-08T20:22:06.3719127Z [INFO] Arquillian Protocol Aggregator ..................... SUCCESS [  0.032 s]
2024-07-08T20:22:06.3720107Z [INFO] Arquillian BOM ..................................... SUCCESS [  1.065 s]
2024-07-08T20:22:06.3720714Z [INFO] Arquillian Core: Implementation Integration Tests .. FAILURE [  2.799 s]
2024-07-08T20:22:06.3721288Z [INFO] Arquillian Core: Common integration tests .......... SKIPPED
2024-07-08T20:22:06.3722111Z [INFO] Arquillian Core: JUnit 4 Integration Tests ......... SKIPPED
2024-07-08T20:22:06.3723001Z [INFO] Arquillian Core: JUnit 5 Integration Tests ......... SKIPPED
2024-07-08T20:22:06.3724058Z [INFO] Arquillian Core: TestNG Integration Tests .......... SKIPPED
```

It also didn't fail for me locally in a Windows VM.

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (starksm64) [reply .github/workflows/integration-tests.yml:12]: What if we just run SE 21 on the integration tests on each platform automatically and have a complete integration-test job that can be run manually that covers the latest LTS and the two previous? The full set is running in 6.5 minutes which is not too bad.

**Review by starksm64 — APPROVED** [review]

**Pull request merged**